langchain4j/src/main/java/dev/langchain4j/model/openai/OpenAiChatModel.java

```diff
@@ -68,8 +71,15 @@ public OpenAiChatModel(String baseUrl,
         timeout = timeout == null ? defaultTimeoutFor(modelName) : timeout;
         maxRetries = maxRetries == null ? 3 : maxRetries;
 
-        this.client = OpenAiClient.builder()
-                .openAiApiKey(apiKey)
+        final OpenAiClient.Builder builder = OpenAiClient.builder();
```

Comment:
please check `AzureOpenAi*Model` classes

Reply:
Hi @langchain4j,

Thank you for pointing that out. Initially, I was trying to use this set of classes, but I came across [this comment](https://github.com/langchain4j/langchain4j/blob/main/langchain4j/src/main/java/dev/langchain4j/model/azure/AzureOpenAiLanguageModel.java#L23) saying function calling is not supported in `AzureOpenAiLanguageModel`, and this is a key part of what I'm trying to accomplish.

This is when I decided to add the missing code to set up my Azure OpenAI along with `OpenAi*Model` classes and create the PR. 

Best,
Fabio

Reply:
@flanza `AzureOpenAiLanguageModel` does not support function calling, but `AzureOpenAiChatModel` does, as written in the comment.

Reply:
Cool! In this case, I will try it and let you know the outcome. If it works smoothly we can close this PR without merging.

Thank you again for pointing this out 👍